### ISSUE TITLE
Log system is useless

### ISSUE THREAD

**Issue by ffdarkpenguin** [HewlettPackard/oneview-redfish-toolkit#151]
The information being logged by the app is useless. It does not tell what was the error nor where it happened. Development and debugging process is terrible to do

**Comment by ricardoas** [comment]
Do we already known when this is happening or is it related to the single case on #152 ?
For that case it is just a missing `{}` on the error message, though...

Should I start a triage for more cases?

**Comment by ffdarkpenguin** [comment]
I'll do a triage on that. By the looks of tox several  errors are returning and empty log. But in order to help development I'd like the error message to have the line where the error occurred. Sometimes just the error message is not enough to find the exception point. Debugging is awful.

**Comment by ffdarkpenguin** [comment]
I opened a new card for looking for log calls without {}, found and fix 4 cases. Even after fix during the tox run several errors still came out blank. Se some samples below:

**Comment by ffdarkpenguin** [comment]
The logging system is not useful for appointing errors: 1. it does not say on which file the error occur. It would be very helpful for development if files/line of the exception where showed. Also in many cases the error message are empty, even after fixing the missing {} on the format function in some log calls. I think that a centralized logging function would make things easier to change/maintain. I noticed that to see where the error occurred we should had called logging.exception (it would add a back-trace to the exception) and not logging.error as we did in all project. Change this now will implicate in changing all files the call the logging system. In order not do do this now to later notice that this was not the best call I thing the logging criteria/system should be decided first.  Some points to consider: logging with back-trace is grate to development, but we want this to final user? Maybe we should define and use other logging levels like debug/exception for development or critical errors with back-trace, and other logging without back-trace for example. Waiting decision on this.

**Comment by ffdarkpenguin** [comment]
This is blocked until:
1. Define new logging method
2. merge all complete issues because this issue will implicate in changing all files in the project that make use of logging facility

**Comment by ricardoas** [comment]
Calls to `logging.exception(message)` inside except seems to have all information we need for debugging. Do they follow logging level rules (do they continue to be printed if I specify log level to be INFO?)?

**Comment by ffdarkpenguin** [comment]
logging.exception solve all debugging problems. If you don't see it as as problem for the final user (see the back-trace messages in the log) . Regarding the lvl of the logging service:
1. exception has the same log level as error. Actually it's just a wrap to : log.error(msg, exc_info=True). The same could be archived just adding exc_info=True and not changing it to log.exception(); The log levels available in python are:
Level 	Numeric value
CRITICAL 	50
ERROR 	40
WARNING 	30
INFO 	20
DEBUG 	10
NOTSET 	0
So log level INFO = 20. When you set a log level you are telling the log system to ignore logs with LOWER level. So INFO(20) will print all exception (same level as error(40))

**Comment by ricardoas** [comment]
Great! Then we better change to `logging.exception()`!

**Comment by ffdarkpenguin** [comment]
Should be blocked until others merged. or going to have a lot of REDO
